src/plot_api/plot_api.js

```diff
@@ -2093,7 +2093,7 @@ Plotly.update = function update(gd, traceUpdate, layoutUpdate, traces) {
     // fill in redraw sequence
     var seq = [];
 
-    if(restyleFlags.fullReplot && relayoutFlags.layoutReplot) {
+    if(restyleFlags.fullReplot || relayoutFlags.layoutReplot) {
```

Comment:
alright, not sure what I was seeing earlier with `layoutReplot` / `fullReplot`, but putting back the `&&` (https://github.com/plotly/plotly.js/pull/1265/commits/b683903b9c44a83826e52506135bde077ddac0d1) doesn't seem to cause any issues now. Perhaps I fixed it in some other way in the meantime...